### PR TITLE
Fix visible and expand locking

### LINKED ISSUE #506: Massive data race detect!!
**Describe the bug**
Im compile my soft with option -race. After start im loock in console 23 event data race

**Device (please complete the following information):**
 - OS: Ubuntu Linux
 - Version 18.04 LTS
 - Go version 1.13.1
 - Fyne version: last stable, public download from github.

### PR DESCRIPTION
**Pull request by andydotxyz**
The remaining races that can be found all have xxxRenderer.Refresh() in them, which we will resolve in 2.0

Fixes #506

### Checklist:

- [x] Tests included. <- go test or go run fyne_demo with `-race`
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: internal/driver/glfw/window.go [hunk 1]
```diff
@@ -398,8 +402,11 @@ func (w *window) Content() fyne.CanvasObject {
 }
 
 func (w *window) SetContent(content fyne.CanvasObject) {
+	w.viewLock.RLock()
+	visible := w.visible
+	w.viewLock.RUnlock()
 	// hide old canvas element
-	if w.visible && w.canvas.Content() != nil {
+	if visible && w.canvas.Content() != nil {
```
Comment (stuartmscott) [inline internal/driver/glfw/window.go:409]: should w.canvas.Content also be captured in a local variable?

Comment (andydotxyz) [reply internal/driver/glfw/window.go:409]: Canvas does it's own locking

Comment (stuartmscott) [reply internal/driver/glfw/window.go:409]: Ok so Content() should be safe, perhaps capture the canvas reference then?

Comment (stuartmscott) [reply internal/driver/glfw/window.go:409]: Also line 354 accesses the canvas content without going through canvas' locking

File: internal/driver/glfw/loop.go [hunk 2]
```diff
@@ -120,8 +120,14 @@ func (d *gLDriver) runGL() {
 					continue
 				}
 
-				if w.shouldExpand {
+				w.viewLock.RLock()
```
Comment (stuartmscott) [inline internal/driver/glfw/loop.go:123]: What if we captured the viewport once at the start of the loop?
```go
w.viewLock.Lock()
v := w.viewport
w.viewLock.Unlock()

if v == nil {
	continue
}
if v.ShouldClose() {
	//...
}
//...
```

Comment (andydotxyz) [reply internal/driver/glfw/loop.go:123]: That means gaining a write lock 60 * windowCount times every second instead of only when windows are closing

**Comment by andydotxyz** [comment]
Fixed the canvas.Content() access

**Pull request merged**